src/server/pfs/server/env.go

```diff
@@ -41,17 +43,33 @@ type Env struct {
 }
 
 func EnvFromServiceEnv(env serviceenv.ServiceEnv, txnEnv *txnenv.TransactionEnv) (*Env, error) {
-	// Setup etcd, object storage, and database clients.
-	objClient, err := obj.NewClient(env.Context(), env.Config().StorageBackend, env.Config().StorageRoot)
-	if err != nil {
-		return nil, err
+	cfg := env.Config()
+
+	// Only setup the new GoCDK bucket if necessary.  The storage layer will prefer it, if it is non-nil.
+	// Disabling GoCDK will completely bypass the entire setup, and revert to the old obj.Client.
+	// This is designed to be a quick fix in the field, if issues are discovered with Go CDK.
+	var objClient obj.Client
+	var bucket *obj.Bucket
+	if cfg.GoCDKEnabled {
```

Comment:
This is where setup of GoCDK is gated.

Reply:
Have you considered an obj.Client adapter for Go CDK? If that is feasible, it seems like it would be a cleaner way to introduce Go CDK into the storage layer. It wouldn't be exactly the same as calling the Go CDK methods directly, but it should test out the stuff we would be concerned about (i.e. setup and general usage).

Reply:
Adapting GoCDK to obj.Client? That wouldn't gain us anything, the work has already been done to depend on GoCDK directly (which is also a better solution long term).  That's already in master.

Part of the motivation for doing this was to have access to the Go CDK interface, so we can do things like Signed URLs, and get sizes of objects during lists.  Just using Go CDK for setup, is all of the costs, and none of the benefits.

All that's left to do at this point is turn our environment variables into Go CDK buckets.  I think Fahad already has some code that turns `obj.URLs` into Go CDK buckets.  I need to sync with him to see how he recommends adapting that to use the environment variables.

Reply:
Okay, I wasn't aware of the change to using the `kv.Store` interface internally. I thought we were still using the `obj.Client` interface and you were going to switch internally based on what was set.